tests/api_fastapi/core_api/routes/public/test_dag_stats.py

```diff
@@ -0,0 +1,431 @@
+            dag_id=dag_3.dag_id,
+            run_id="test_dag_3_run_id_1",
+            run_type=DagRunType.MANUAL,
+            execution_date=timezone.parse(self.default_time),
+            start_date=timezone.parse(self.default_time),
+            external_trigger=True,
+            state="success",
+        )
+        entities = (
+            dag_1,
+            dag_1_run_1,
+            dag_1_run_2,
+            dag_2,
+            dag_2_run_1,
+            dag_3,
+            dag_3_run_1,
+        )
+        for entity in entities:
+            session.add(entity)
+        session.commit()
+
+    @pytest.fixture(autouse=True)
+    def setup(self) -> None:
+        self._clear_db()
+
+    def teardown_method(self) -> None:
+        self._clear_db()
+
+
+class TestGetDagStats(TestDagStatsEndpoint):
+    """Unit tests for Get DAG Stats."""
+
+    def test_should_respond_200(self, client, session):
+        self._create_dag_and_runs(session)
+        exp_payload = {
+            "dags": [
+                {
+                    "dag_id": DAG1_ID,
+                    "stats": [
+                        {
+                            "state": DagRunState.QUEUED,
+                            "count": 0,
+                        },
+                        {
+                            "state": DagRunState.RUNNING,
+                            "count": 1,
+                        },
+                        {
+                            "state": DagRunState.SUCCESS,
+                            "count": 0,
+                        },
+                        {
+                            "state": DagRunState.FAILED,
+                            "count": 1,
+                        },
+                    ],
+                },
+                {
+                    "dag_id": DAG2_ID,
+                    "stats": [
+                        {
+                            "state": DagRunState.QUEUED,
+                            "count": 1,
+                        },
+                        {
+                            "state": DagRunState.RUNNING,
+                            "count": 0,
+                        },
+                        {
+                            "state": DagRunState.SUCCESS,
+                            "count": 0,
+                        },
+                        {
+                            "state": DagRunState.FAILED,
+                            "count": 0,
+                        },
+                    ],
+                },
+            ],
+            "total_entries": 2,
+        }
+
+        response = client().get(f"{API_PREFIX}?dag_ids={DAG1_ID},{DAG2_ID}")
+        assert response.status_code == 200
+        res_json = response.json()
+        assert len(res_json["dags"]) == 2
+        assert sorted(
+            res_json["dags"],
+            key=lambda d: d["dag_id"],
+        ) == sorted(
+            exp_payload["dags"],
+            key=lambda d: d["dag_id"],
+        )
```

Comment:
Why are we manually sorting the dag_id ? This should be enforced by the endpoint. A stable order should be returned, that we can test.

Reply:
Thanks, updated PR. Not sure why I didn't notice and remove this extra sorting! 🥲